consult.el

```diff
@@ -3010,48 +3015,72 @@ INITIAL is the initial input."
-    (unless (use-region-p)
-      (goto-char (point-min)))
-    (lambda (input restore)
-      ;; New input provided -> Update
-      (when (and input (not (equal input last-input)))
-        (if (string-match-p "\\`!? ?\\'" input)
-            ;; Special case the empty input for performance.
-            (progn
-              (dolist (ov overlays)
-                (overlay-put ov 'invisible nil))
-              (setq last-input input))
-          (let* ((not (string-prefix-p "! " input))
-                 (stripped (string-remove-prefix "! " input))
-                 ;; Heavy computation is interruptible if *not* committing!
-                 (ht (if restore
-                         (consult--string-hash (funcall filter stripped lines))
-                       (while-no-input
-                         (consult--string-hash (funcall filter stripped lines))))))
-            (when (hash-table-p ht)
-              (let ((ov overlays) (li lines))
-                (while ov
-                  (overlay-put (car ov) 'invisible (eq not (gethash (car li) ht)))
-                  (setq li (cdr li) ov (cdr ov))))
-              (setq last-input input)))))
-      (when restore
-        (cond
-         ((not input)
-          (goto-char point-orig))
-         ((equal input "")
-          (consult-focus-lines 'show))
-         (t
-          ;; Sucessfully terminated -> Remember invisible overlays
-          (dolist (ov overlays)
-            (if (overlay-get ov 'invisible)
-                (push ov consult--focus-lines-overlays)
-              (delete-overlay ov)))
-          (setq overlays nil)))
-        ;; Destroy remaining overlays
-        (mapc #'delete-overlay overlays)))))
+        (let ((i 0) line)
+	  (consult--each-line beg end
+	    (setq line
+		  (if (eq beg end) (char-to-string ?\n) ; "\n" reuses string! 
+		    (buffer-substring-no-properties beg end)))
+	    (add-text-properties 0 1 `(line (,(cl-incf i) ,beg ,end)) line)
+	    (push line lines)))))
+    (cl-labels ((add-inv-ov (beg end)
+			    (push (consult--overlay beg end 'invisible t)
+				  overlays)))
+      (lambda (input restore)
+	;; New input provided -> Update
+	(when (and input (not (equal input last-input)))
+	  (mapc #'delete-overlay overlays)
+	  (setq last-input input overlays nil)
+          (unless (string-match-p "\\`!? ?\\'" input) ; empty input.
+            (let* ((not (string-prefix-p "! " input))
+                   (stripped (string-remove-prefix "! " input))
+                   ;; Heavy computation is interruptible if *not* committing!
+                   (matches (if restore
+				(consult--extract-line-ind
+				 (funcall filter stripped lines))
+			      (while-no-input
+				(consult--extract-line-ind
+				 (funcall filter stripped lines)))))
+		   (old-ind 0)
+		   (start (point-min))
+		   (finish nil)
+		   (max (point-max)))
+	      (unless (eq matches t)	;input arrived
+		(message "%s %d lines" (if not "Excluded" "Included") (length matches))
```

Comment:
Please remove this message. We can discuss this separately, but I'd like to have the changes at least as independent commits.

Reply:
OK figured, but thought I'd prime the pump.  The better solution of course would be a consult-buffer style prefix of `#lines shown/#total lines`, but I didn't know how to do that.

Reply:
Yes, I would accept a PR which adds such an overlay for `consult-focus-lines` and `consult-keep-lines`. But it should be a separate PR.